activation/activation.go

```diff
@@ -507,14 +531,28 @@ func (b *Builder) run(ctx context.Context, sig *signing.EdSigner) {
 
 func (b *Builder) BuildNIPostChallenge(ctx context.Context, nodeID types.NodeID) (*types.NIPostChallenge, error) {
 	logger := b.logger.With(log.ZShortStringer("smesherID", nodeID))
+
+	if err := b.identitiesStates.Set(nodeID, types.IdentityStateWaitForATXSyncing); err != nil {
+		b.logger.Warn("failed to switch identity state",
+			zap.Stringer("smesherID", nodeID),
+			zap.Error(err),
+		)
+	}
+
 	select {
 	case <-ctx.Done():
 		return nil, ctx.Err()
 	case <-b.syncer.RegisterForATXSynced():
 	}
-
```

Comment:
`BuildNIPostChallenge` is called right after the start of the node and after publishing an ATX. Only in the first case (node just started) the "atx syncing" state will be active for more than a few ms.

Maybe it makes sense to move this check up the call chain into the `run` method before the `for` loop that calls `PublishActivationTx` (around line 466)?

Reply:
Will check, got it